### PR TITLE
EDU-811 Add flush function to debounced callback hook

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-811

### CODE REVIEW

File: src/ui/hooks.js [hunk 1]
```diff
@@ -38,21 +39,33 @@ export function useDebouncedCallback(callback, timeLimit = 250) {
...
+      lastPendingArgs.current = null;
     }, timeLimit);
   }, [timeLimit, storedCallback]);
+
+  if (!debouncedCallback.flush) {
+    debouncedCallback.flush = function flush() {
+      timeout.current && clearTimeout(timeout.current);
+      timeout.current = null;
+      lastPendingArgs.current && storedCallback.current(...lastPendingArgs.current);
+      lastPendingArgs.current = null;
```
Comment (adrianaluchian) [inline src/ui/hooks.js:64]: To make the code a bit easier to follow and reduce this repetition in all 3 cases, maybe you could extract lines 63-64  into a `executePendingCallback` method. I think the name would help.

**Comment by adrianaluchian** [comment]
Super nice solution! 👏

File: src/ui/hooks.js (hunk 1 above)
Comment (ahelmberger) [reply src/ui/hooks.js]: Good idea, extract them (and also the two lines above) into separate functions.

Comment (adrianaluchian) [reply src/ui/hooks.js]: 👍

**Pull request merged**